### PR TITLE
Use `reject` with `TransactionPollingTimeoutError`

### PR DESCRIPTION
**Pull request by Muhammad-Altabba**
## Description

Fixes #([issue](https://github.com/ChainSafe/web3.js/issues/5334))

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` with success.
- [ ] I have tested the built `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

File: packages/web3-eth/src/utils/wait_for_transaction_receipt.ts [hunk 1]
```diff
@@ -27,20 +27,23 @@ export async function waitForTransactionReceipt<ReturnFormat extends DataFormat>
 	transactionHash: Bytes,
 	returnFormat: ReturnFormat,
 ): Promise<TransactionReceipt> {
-	return new Promise(resolve => {
+	const pollingInterval =
+		web3Context.transactionReceiptPollingInterval ?? web3Context.transactionPollingInterval;
```
Comment (jdevcs) [inline packages/web3-eth/src/utils/wait_for_transaction_receipt.ts:31]: can you add a test for this function , by using higher nonce and modified `transactionReceiptPollingInterval` ( to few second ) and look for promise reject after few seconds.

**Review by jdevcs — CHANGES_REQUESTED** [review]

File: packages/web3-eth/src/utils/wait_for_transaction_receipt.ts (hunk 1 above)
Comment (Muhammad-Altabba) [reply packages/web3-eth/src/utils/wait_for_transaction_receipt.ts:31]: Actually raising the nonce revealed 3 issues. And in this MR I am addressing only one of them to make it easier to do the review and merge... and as this issue is somehow separated from others. 
And in a different MR (https://github.com/ChainSafe/web3.js/pull/5311) the nonce is raised in the integration test to validate handling all cases that a provider could respond... :smile:

Comment (jdevcs) [reply packages/web3-eth/src/utils/wait_for_transaction_receipt.ts:31]: sure you can add test case for this in 5311, good work for finding this issue.

**Review by jdevcs — APPROVED** [review]

File: packages/web3-eth/src/utils/wait_for_transaction_receipt.ts [hunk 2]
```diff
@@ -27,20 +27,23 @@ export async function waitForTransactionReceipt<ReturnFormat extends DataFormat>
...
-					web3Context.transactionPollingInterval;
+				transactionPollingDuration += pollingInterval;
 
 				if (transactionPollingDuration >= web3Context.transactionPollingTimeout) {
 					clearInterval(intervalId);
-					throw new TransactionPollingTimeoutError({
-						numberOfSeconds: web3Context.transactionPollingTimeout / 1000,
-						transactionHash,
-					});
+					reject(
```
Comment (nazarhussain) [inline packages/web3-eth/src/utils/wait_for_transaction_receipt.ts:40]: 👍

**Review by nazarhussain — APPROVED** [review]

**Pull request merged**